.github/kubeconform.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+set -euo pipefail
+
+KUBECONFORM_VERSION="v0.6.3"
+# https://github.com/yannh/kubeconform/issues/51
+CRD_SPEC_URL="https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/v3/apis__apiextensions.k8s.io__v1_openapi.json"
+OS=$(uname)
+
+CHANGED_CHARTS=${CHANGED_CHARTS:-${1:-}}
+if [ -n "$CHANGED_CHARTS" ];
+then
+  CHART_DIRS=$CHANGED_CHARTS
+else
+  CHART_DIRS=$(ls -d charts/*)
+fi
+
+# install kubeconform
+curl --silent --show-error --fail --location --output /tmp/kubeconform.tar.gz "https://github.com/yannh/kubeconform/releases/download/${KUBECONFORM_VERSION}/kubeconform-${OS}-amd64.tar.gz"
+tar -xf /tmp/kubeconform.tar.gz kubeconform
+
+# validate charts
+for CHART_DIR in ${CHART_DIRS}; do
+  echo "Running kubeconform for folder: '$CHART_DIR'"
+
+  # Note: -ignore-missing-schemas could be added if needed, but not currently
+  # needed since we have the schema necessary to validate the CRDs themselves.
+  #
+  # Also, if at some point we needed to validate things _using_ these CRDs,
+  # they're available via
+  # https://github.com/datreeio/CRDs-catalog/tree/main/datadoghq.com
+  helm dep up "${CHART_DIR}" && helm template --kube-version "${KUBERNETES_VERSION#v}" \
+        --values "${CHART_DIR}"/ci/kubeconform-values.yaml "${CHART_DIR}" \
+    | ./kubeconform -schema-location default -schema-location "$CRD_SPEC_URL" \
+        -output pretty -verbose -kubernetes-version "${KUBERNETES_VERSION#v}" -
```

Comment:
```suggestion
        -output pretty -strict -verbose -kubernetes-version "${KUBERNETES_VERSION#v}" -
```
`-strict` was enabled with `kubeval`, any reason not to enable it with `kubeconform`?

Reply:
it has been a while since I put this together / tested it, but it looks like this is probably why:
```
✖ stdin: DaemonSet failed validation: release-name-datadog error unmarshalling resource: error converting YAML to JSON: yaml: unmarshal errors:
  line 879: key "kubernetes.io/os" already set in map
✖ stdin: Deployment failed validation: release-name-datadog-clusterchecks error unmarshalling resource: error converting YAML to JSON: yaml: unmarshal errors:
  line 181: key "kubernetes.io/os" already set in map
✖ stdin: Deployment failed validation: release-name-datadog-cluster-agent error unmarshalling resource: error converting YAML to JSON: yaml: unmarshal errors:
  line 242: key "kubernetes.io/os" already set in map
```

and in fact, this happens a bunch in the configs:
```yaml
      nodeSelector:
        kubernetes.io/os: linux
        kubernetes.io/os: linux
```

My suggestion is that we don't try to fix that in this PR, but I can rebase if someone wants to throw up a fix for it?

Reply:
I'm fine with that, up to the maintainers of these charts 👍

Reply:
Are you able to bug someone about it? I can try to take a quick look and see if there's an obvious fix, but it would nice to be able to enable strict here vs. waiting, since it's kind of buried here.

I checked, and it's only `charts/datadog` that has the issue - all the others succeed with `-strict`. If it's a really easy fix, I can try to open up a PR for that

Reply:
```
      nodeSelector:
        {{ template "label.os" . }}: {{ .Values.targetSystem }}
      {{- if .Values.agents.nodeSelector }}
{{ toYaml .Values.agents.nodeSelector | indent 8 }}
      {{- end }}
```

Maybe because of this (and similar code in other sections)?

Reply:
I created #1312 

One possibility is that the chart _intends_ to allow multiple `nodeSelector` statements, and that this isn't intended to override the baked in defaults? It looks like a feature was released in 2.x to always includle the target system nodeSelector; in that case, there's no real bug, and the "fix" would probably be to just use some kind of arbitrary `nodeSelector` value that doesn't match in this PR. I'm happy to make that change if we can confirm this is the intended behavior.

Reply:
Indeed it worked